### PR TITLE
feat: Template XML and JSON Files

### PR DESCRIPTION
**Pull request by JohJakob**
XML and JSON files can now include template tags and will be compiled. This allows for dynamically generated RSS/JSON feeds and XML Sitemaps.

For now, the template compiler processes files with an `.xml`, `.rss` or `.json` extension the same way it processes HTML files. This works fine as long as you do not use content types that output HTML.

This functionality can be improved by modifying the template compiler to recognize XML and JSON files and output some content types differently (e. g. render the Image content type without the `img` tag by default, so one does not have to use the `tag=false` parameter).

I have tested it locally with an XML and a JSON file. Should I add a proper test? The changes are not huge though.

### CODE REVIEW

**Comment by JohJakob** [comment]
Commit [9527900](https://github.com/vapid/vapid/pull/177/commits/9527900010333ddc8548d3b45af29cba03f295b8) also makes Vapid serve static XML and JSON files with their corresponding Content-Type that are currently served with the `text/html` Content-Type and rendered as HTML pages in the browser.

**Comment by JohJakob** [comment]
Yes, this one is ready for review.
I’ll try to write a test for it.

**Comment by JohJakob** [comment]
@srobbin Could you help me out a bit or point me in the right direction? I’m not quite sure what tests to add.

**Comment by srobbin** [comment]
@JohJakob I'll take a look and see if I can provide some guidance or a test.

**Comment by JohJakob** [comment]
@srobbin Thank you :)

**Comment by JohJakob** [comment]
@srobbin Ah, yes! I’ll take a look at that and add a check so livereload will only be used in HTML files.

**Comment by JohJakob** [comment]
@srobbin The livereload script should only be injected into HTML files now ([see this commit](https://github.com/vapid/vapid/pull/177/commits/116c14fd58dcc9fb2d48dbcbb4c5ddfaf2ba6cac)).

File: lib/runners/VapidServer/index.js [hunk 1]
```diff
@@ -80,6 +80,9 @@ class VapidServer extends Vapid {
       const ext = path.extname(ctx.path);
 
       switch (ext) {
+        case '':
+          ctx.type = 'text/html';
+          break;
```
Comment (srobbin) [inline lib/runners/VapidServer/index.js:85]: @JohJakob What's the thought here, with an empty case? Are there times where the extension is blank?

Comment (JohJakob) [reply lib/runners/VapidServer/index.js:85]: @srobbin Yes, this accounts for all index.html files that don’t get served with their full path.

**Comment by JohJakob** [comment]
@srobbin Do you think this requires any additional tests? It works the same way as in HTML files, just with other plain text file types.

**Comment by srobbin** [comment]
@JohJakob I'm sorry, I haven't been maintaining this repo. I'll merge this in now. Thank you again for adding this feature—it's great.

**Comment by JohJakob** [comment]
@srobbin No worries! It wasn’t urgent. Thanks for merging!

**Comment by JohJakob** [comment]
@srobbin Could you also add a new release with this feature, please? 🙂

**Comment by srobbin** [comment]
@JohJakob Of course, I'm working on that now. It should be up shortly.

**Comment by JohJakob** [comment]
@srobbin Ah, great! Thank you so much!

**Pull request merged**